### DISCUSSION TITLE
How to use xml namespace feature but make ts not throw error?

### DISCUSSION THREAD

**Discussion by zhaoyao91** [solidjs/solid#395] [Q&A] [answered]
Solid support xml namespace, it's very nice, but ts error me ` Property 'on:beforeinput' does not exist on type 'InputHTMLAttributes<HTMLInputElement>'.`  I have to `// @ts-ignore` the line but this is not nice. Is there a standard way to support use namespace feature in ts project?

**Comment by ryansolid** [comment, marked as answer]
Yes I need to move into general docs: https://github.com/ryansolid/solid/blob/main/CHANGELOG.md#namespace-types

**Reply by zhaoyao91** [reply]
The changelog says we should add these types to our project. Could them be included in solid directly?

**Reply by ryansolid** [reply]
No because all I could do would be alias to something generic. And I can't even do that until at least TS 4.3 since they don't allow tag literal types in index signatures currently.
